gomock/call.go

```diff
@@ -75,6 +75,9 @@ func (c *Call) Do(f interface{}) *Call {
 	return c
 }
 
+// Return sets the values to return when the target method is called.
+//
+// If Return and Do are called, the values returned via Do will be returned.
```

Comment:
That's actually exactly the style I was considering for my own mock system, with each function in the interface being mocked having a method on the mock struct, which calls a function field, and that function field can be replaced as desired by either named functions or anonymous enclosures.